cookbooks/aws-parallelcluster-install/spec/unit/recipes/awscli_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'spec_helper'
+
+describe 'aws-parallelcluster-install::awscli' do
+  before do
+    mock_exist_call_original
+  end
+
+  for_all_oses do |platform, version|
+    let(:chef_run) do
+      ChefSpec::Runner.new(platform: platform, version: version)
+    end
+
+    context 'when aws cli exists' do
+      before do
+        mock_file_exists("/usr/local/bin/aws", true)
+        chef_run.converge(described_recipe)
+      end
+
+      it "should not install it" do
+        is_expected.not_to run_bash("install awscli")
+      end
+    end
+
+    context 'when aws cli does not exist' do
+      before do
+        mock_file_exists("/usr/local/bin/aws", false)
+        chef_run.converge(described_recipe)
+      end
+
+      it "should install unzip" do
+        is_expected.to install_package("unzip")
+      end
+
+      it "should install awscli" do
+        is_expected.to run_bash("install awscli")
```

Comment:
is there a way to make chefSpec test the bash code also?

Reply:
Yes but I avoided it intentionally.
We test the outcome of that bash code via kitchen tests. 
I don't see value in testing that specific way of installing aws cli.
If we refactor it in the future using chef resources it might make more sense to test it.